ocaml/xapi/message_forwarding.ml

```diff
@@ -1541,6 +1541,11 @@ module Forward = functor(Local: Custom_actions.CUSTOM_ACTIONS) -> struct
         then vm
         else Xapi_vm_snapshot.create_vm_from_snapshot ~__context ~snapshot in
 
+      (* As when vm snapshot is not allowed if vm has VUSBs, so revert should be disallowed *)
```

Comment:
I wonder whether this could fail earlier to avoid creating the VM first only to fail now. But it is probably not worth the complexity. This looks straight forward to me.

Reply:
yes, I think fail here is reasonable.

Reply:
Not convinced (yet...!)

Reply:
oh, how do you think about this? @jonludlam

Reply:
OK, what is the difficulty with putting this test before the create_vm_from_snapshot?

Reply:
If vm is not valid , how can we get the VUSBs?